src/main/java/com/dev/cinema/config/SecurityConfig.java

```diff
@@ -1,27 +1,38 @@
 package com.dev.cinema.config;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder encoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder encoder) {
+        this.userDetailsService = userDetailsService;
+        this.encoder = encoder;
+    }
+
     @Autowired
     public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
-        auth.inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("user@gmail.com").password(getEncoder().encode("1234")).roles("USER");
+        auth
+                .userDetailsService(userDetailsService)
+                .passwordEncoder(encoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
-        http.authorizeRequests()
-                .anyRequest()
-                .authenticated()
+        http
+                .authorizeRequests()
+                .antMatchers("/register").permitAll()
+                .antMatchers(HttpMethod.POST,"/movies/").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET,"/movies/").hasRole("USER")
```

Comment:
Authorised user will have access to GET methods without it , so you can delete it